### PR TITLE
Upgrade to rails 4.2.x

### PR DESCRIPTION
**Pull request by 3v0k4**

### CODE REVIEW

File: app/models/measurement.rb [hunk 1]
```diff
@@ -24,7 +24,6 @@ class Measurement < ActiveRecord::Base
   Y_SIZES = (1..300).map { |i| 1.2 ** i * 0.000001 }
   FIXED_MEASUREMENTS_IN_A_DAY = 1440
 
-  # belongs_to :session, :through => :stream, :inverse_of => :measurements, :counter_cache => true
```
Comment (AkkarinForest) [inline app/models/measurement.rb]: We don't need this association anymore? Can it influence performance?

File: app/models/user.rb [hunk 2]
```diff
@@ -111,4 +95,13 @@ def admin?
...
     self.username.chomp!
   end
+
+  # TokenAuthenticatable was removed from Devise in 3.1
+  # https://gist.github.com/josevalim/fb706b1e933ef01e4fb6
+  def generate_authentication_token
+    loop do
+      token = Devise.friendly_token
+      break token unless User.where(authentication_token: token).first
+    end
```
Comment (AkkarinForest) [inline app/models/user.rb]: Are we sure that this always breaks? I mean is there a risk of infinite loop?

File: Gemfile [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
+gem 'sass-rails'
+gem 'coffee-rails'
 
-# Gems used only for assets and not required
-# in production environments by default.
-group :assets, :development do
-  gem 'coffee-rails', '~> 3.2.1'
-  gem 'sass-rails', '~> 3.2.2'
-  gem 'uglifier', '~> 1.2.7'
+group :development do
```
Comment (AkkarinForest) [inline Gemfile]: Any reason we need two development groups? there is another one on line 40

File: db/migrate/20190212125754_change_collation_for_tag_names.acts_as_taggable_on_engine.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This migration comes from acts_as_taggable_on_engine (originally 5)
+# This migration is added to circumvent issue #623 and have special characters
+# work properly
+class ChangeCollationForTagNames < ActiveRecord::Migration
+  def up
+    if ActsAsTaggableOn::Utils.using_mysql?
+      execute("ALTER TABLE tags MODIFY name varchar(255) CHARACTER SET utf8 COLLATE utf8_bin;")
+    end
+  end
+end
```
Comment (AkkarinForest) [inline db/migrate/20190212125754_change_collation_for_tag_names.acts_as_taggable_on_engine.rb]: Should this migration also have a `down` method?

File: lib/session_builder.rb [hunk 5]
```diff
@@ -24,10 +24,10 @@ def build_session!
     data = build_local_start_and_end_time(data)
 
     begin
-      session = Session.create!(data)
+      session = Session.create!(data.reject { |k| [:drawable, :deleted].include?(k)})
```
Comment (AkkarinForest) [inline lib/session_builder.rb]: What is a :drawable session? I'm just curious, cause I've never seen that.

File: spec/models/measurement_spec.rb [hunk 6]
```diff
@@ -16,14 +16,9 @@
...
 # You can contact the authors by email at <info@habitatmap.org>
 
-require 'spec_helper'
+require 'rails_helper'
 
 describe Measurement do
-  it { is_expected.to validate_presence_of :value }
-  it { is_expected.to validate_presence_of :longitude }
-  it { is_expected.to validate_presence_of :latitude }
-  it { is_expected.to validate_presence_of :time }
```
Comment (AkkarinForest) [inline spec/models/measurement_spec.rb]: Have you consider rewriting them to something like 
```
 it "is invalid without a value" do
   m = Measurement.new(value: nil)
   m.valid?
   expect(m.errors[:value]).to include("can't be blank")
 end
```
Or maybe we don't need to test validations in newer rails versions?

File: app/controllers/api/measurement_sessions_controller.rb [hunk 7]
```diff
@@ -20,6 +20,9 @@ module Api
   class MeasurementSessionsController < BaseController
     INT_Q_ATTRS = [:time_from, :time_to, :day_from, :day_to]
 
+    # TokenAuthenticatable was removed from Devise in 3.1
+    # https://gist.github.com/josevalim/fb706b1e933ef01e4fb6
+    before_filter :authenticate_user_from_token!, :only => :create
     before_filter :authenticate_user!, :only => :create
```
[the remark this answers is not in the document]
Comment (3v0k4) [reply app/controllers/api/measurement_sessions_controller.rb]: Devise removed TokenAuthenticable which was in charge of doing what I've added in this PR. Each user when created has a token (string) which is sent with each request to authenticate itself.

`authenticate_user_from_token!` tries to authenticate the user looking for the token. `authenticate_user!` tries to authenticate in a different way. That's more or less how I understand it

File: app/models/measurement.rb (hunk 1 above)
Comment (3v0k4) [reply app/models/measurement.rb]: It was already commented out. Just removed the comment ;)

File: app/models/user.rb (hunk 2 above)
Comment (3v0k4) [reply app/models/user.rb]: Copied from the Devise wiki. I trust that

File: Gemfile (hunk 3 above)
Comment (3v0k4) [reply Gemfile]: Good catch! Fixed it

File: db/migrate/20190212125754_change_collation_for_tag_names.acts_as_taggable_on_engine.rb (hunk 4 above)
Comment (3v0k4) [reply db/migrate/20190212125754_change_collation_for_tag_names.acts_as_taggable_on_engine.rb]: Not sure really. It's generated by the gem

File: lib/session_builder.rb (hunk 5 above)
Comment (3v0k4) [reply lib/session_builder.rb]: With Rails 4 trying to create a model w/ an attribute that doesn't exist on the model fails. 
```
User.create(pizza: "pants") // BOOM
```
Those two are coming from the mobile app.

This should be handled in a better way (selecting what's right and not removing what's wrong). But hopefully somebody else will stumble on this and improve

File: spec/models/measurement_spec.rb (hunk 6 above)
Comment (3v0k4) [reply spec/models/measurement_spec.rb]: I think these tests are meaningless. The fact that `validates :....` works that way is tested by rails. I believe declarative code of a framework should not be tested.

It's like testing that 1 + 1 == 1. I trust ruby to give me a working `+`

**Pull request merged**